ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -265,10 +265,26 @@ MemberFunction::MemberFunction(HqlCppTranslator & _translator, BuildCtx & classc
 MemberFunction::MemberFunction(HqlCppTranslator & _translator, BuildCtx & classctx, const char * text, unsigned _flags) : translator(_translator), ctx(classctx), flags(_flags)
 {
     stmt = ctx.addQuotedFunction(text, (flags & MFdynamicproto) != 0);
+    if (flags & MFoptimize)
+        stmt->setOptimize(true);
+    else if (flags & MFnooptimize)
+        stmt->setOptimize(false);
 }
 
 MemberFunction::~MemberFunction() noexcept(false)
 {
+    unsigned maxOptimizeSize = translator.queryOptions().maxOptimizeSize;
+    unsigned minNoOptimizeSize = translator.queryOptions().minNoOptimizeSize;
+    if (stmt && (maxOptimizeSize || minNoOptimizeSize))
+    {
+        //This estimate is before the peephole optimize is applied, so may be a large over estimate
+        unsigned estimatedSize = calcTotalChildren(stmt);
+        if (maxOptimizeSize && (estimatedSize <= maxOptimizeSize))
+            stmt->setOptimize(true);
```

Comment:
It looks like this overrides the optimize flag provided in the constructor.  So, if define this:
`MemberFunction  x(...,MFnooptimize); `

then here it forces the function to optimize.

Reply:
If estimateSize>maxOptimizeSize, then it does change the value for optimize here - it does not set it to false. This means the value of optimize will be the value as set in the  MemberFunction's constructor (lines 268-269).  i.e. if the flag & optimize=true, the optimize will not be set to false even if the estimated size is too big.